packages/firestore/src/local/indexeddb_persistence.ts

```diff
@@ -421,7 +435,36 @@ export class IndexedDbPersistence implements Persistence {
             return this.acquireOrExtendPrimaryLease(txn).next(() => result);
           });
       } else {
-        return transactionOperation(txn);
+        return this.verifyAllowTabSynchronization(txn).next(() =>
+          transactionOperation(txn)
+        );
+      }
+    });
+  }
+
+  /**
+   * Verifies that the current tab is the primary leaseholder or alternatively
+   * that the leaseholder has opted into multi-tab synchronization.
+   */
+  // TODO(multitab): Remove this check when `synchronizeTabs` can no longer be
+  // turned off.
+  private verifyAllowTabSynchronization(
+    txn: SimpleDbTransaction
+  ): PersistencePromise<void> {
+    const store = ownerStore(txn);
+    return store.get('owner').next(currentPrimary => {
+      const currentLeaseIsValid =
+        currentPrimary !== null &&
+        this.isWithinMaxAge(currentPrimary.leaseTimestampMs) &&
+        currentPrimary.ownerId !== this.getZombiedClientId();
+
+      if (currentLeaseIsValid && !this.isLocalClient(currentPrimary)) {
+        if (!currentPrimary.allowTabSynchronization) {
```

Comment:
Intentionally using nested-if over `&&` ?

Reply:
To me, this is easier to follow since it extracts the more important part into a separate statement.